_posts/2019-03-14-2019-kafka-connector.md

```diff
@@ -0,0 +1,183 @@
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: kafka
+        component: kafka-connector
+    spec:
+      containers:
+      - name: kafka
+        image: openfaas/kafka-connector:0.4.0
+        env:
+          - name: gateway_url
+            value: "http://gateway.openfaas:8080"
+          - name: topics
+            value: "faas-request,"
+          - name: print_response
+            value: "true"
+          - name: print_response_body
+            value: "true"
+          - name: basic_auth
+            value: "true"
+          - name: secret_mount_path
+            value: "/var/secrets/"
+          - name: kafka_broker
+            value: "kafka"
+        volumeMounts:
+            - name: auth
+              readOnly: true
+              mountPath: "/var/secrets/"
+      volumes:
+      - name: auth
+        secret:
+          secretName: basic-auth
+```
+
+If you deployed your OpenFaaS instance with Helm and default values there is no need to make changes to this file. The `kafka_broker` variable is our broker host from which we will pull topics.
+
+In our case the broker host is located in service called `kafka` and it is in the same namespace as the gateway plus it uses the default kafka port `9092`.
+
+The environmental variable `topics` is the place in which we list the topics in which we are interested to receive from the broker.
+
+## Deploy the components
+
+In order to deploy the whole stack and play around with it go ahead and apply the kubernetes files while inside the kafka-connector folder:
+
+```bash
+$ kubectl apply -f yaml/kubernetes
+```
+
+We will deploy simple producer provided by Alex:
+
+```bash
+$ kubectl run -n openfaas -t -i --image alexellis2/kafka-producer:0.2.0 producer
+```
+
+Congrats now you have the whole stack deployed:
+* Zookeeper
+* Producer Service
+* Broker Host
+* Kafka Connector
+
+Before we proceed, make sure you have all the components running:
+
+```bash
+$ kubectl get pods -n openfaas
+NAME                               READY   STATUS    RESTARTS   AGE
+alertmanager-b55c578d4-87dgf       1/1     Running   0          37m
+faas-idler-7945c9bc7b-74z68        1/1     Running   1          37m
+gateway-7b7455dfd8-kzktf           2/2     Running   1          37m
+kafka-broker-544fbccd48-b5f65      1/1     Running   0          33m
+kafka-connector-7898548745-5wclq   1/1     Running   0          33m
+nats-6686bb4b95-lj6b8              1/1     Running   0          37m
+producer-c77fb7ccf-cx8xg           1/1     Running   0          31m
+prometheus-7dd5d65568-bkr8j        1/1     Running   0          37m
+queue-worker-8655bb54cb-q5nk8      1/1     Running   1          37m
+zookeeper-699f568f6f-wpzb6         1/1     Running   0          33m
+
+```
+
+## Subscribe function
+
+In order to consume topics from the connector we should just apply annotation with key `topic` and value whichever you used in the `connector-dep.yml` in our case it was `faas-request`.
```

Comment:
In order to consume topics via the connector we simply need to apply an annotation with a key of `topic` and the chosen value used in the `connector-dep.yml` - in our case it was `faas-request`.

What if there are multiple topics stated in `connector-dep.yml`?  Can a function subscribe to multiple topics?

Reply:
Added how to subscribe to multiple topics even though explicitly it is not supported 👍

Reply:
Just to capture what we discussed on Slack, and for the benefit of others viewing this, we've established that observing multiple topics via the connector is a supported feature.